src/toxcore/onion/packet/onion_data_response.rs

```diff
@@ -199,19 +199,22 @@ impl OnionDataResponsePayload {
 pub enum OnionDataResponseInnerPayload {
     /// [`DhtPkAnnouncePayload`](../../dht/packet/struct.DhtPkAnnouncePayload.html) structure.
     DhtPkAnnounce(DhtPkAnnouncePayload),
-    // TODO: FriendRequest
+    /// [`FriendRequest`](../../dht/packet/struct.FriendRequest.html) structure.
+    FriendRequest(FriendRequest),
```

Comment:
Should not we add tests for OnionDataResponseInnerPayload with FriendRequest, at least a successful condition? (example of unsuccessful for DhtPkAnnounce in onion_data_response_payload_encrypt_decrypt_invalid_key)

Reply:
done